aquadoggo/src/graphql/types/paginated_response.rs

```diff
@@ -4,27 +4,43 @@ use async_graphql::dynamic::{Field, FieldFuture, FieldValue, Object, TypeRef};
 use async_graphql::Value;
 use p2panda_rs::schema::Schema;
 
+use crate::db::query::Cursor;
 use crate::graphql::constants;
+use crate::graphql::types::DocumentValue;
 use crate::graphql::utils::{downcast_document, paginated_document_name, paginated_response_name};
 
 /// Pagination data passed from parent to child query fields.
 #[derive(Default, Clone, Debug)]
-pub struct PaginationData {
-    /// The total page count.
-    total_count: u64,
+pub struct PaginationData<C>
+where
+    C: Cursor,
+{
+    /// Number of all documents in queried collection.
+    pub total_count: u64,
 
-    /// Whether this query has a next page waiting.
-    has_next_page: bool,
+    /// Flag indicating if `endCursor` will return another page.
+    pub has_next_page: bool,
+
+    /// Flag indicating if `startCursor` will return another page.
+    pub has_previous_page: bool,
+
+    /// Cursor which can be used to paginate backwards.
+    pub start_cursor: C,
+
+    /// Cursor which can be used to paginate forwards.
+    pub end_cursor: C,
```

Comment:
I'd like to introduce these fields in the GraphQL API as well to be closer to the connection specification but also to allow backwards pagination and have the convenience of getting the crucial cursors directly from the page info, not from the paginated items.